pkg/watcher/internal/metricsprovider/signalfx.go

```diff
@@ -206,3 +279,110 @@ func decodeMetricsPayload(payload interface{}) (float64, error) {
 	}
 	return timestampUtilisation[1].(float64), nil
 }
+
+/**
+Sample metricData payload:
+{
+  "data": {
+    "Ehql_bxBgAc": [
+      [
+        1600213380000,
+        84.64246793530153
+      ]
+    ],
+    "EuXgJm7BkAA": [
+	  [
+		1614634260000,
+		5.450946379084264
+     ]
+    ],
+    ....
+    ....
+  },
+  "errors": []
+}
+
+https://dev.splunk.com/observability/reference/api/metrics_metadata/latest#endpoint-retrieve-metric-timeseries-metadata
+Sample metaData payload:
+{
+    "count": 5,
+    "partialCount": false,
+    "results": [
+        {
+            "active": true,
+            "created": 1614534848000,
+            "creator": null,
+            "dimensions": {
+                "host": "gpma392019.gpf-prod.us-central1.gcp.dev.paypalinc.com",
+                "sf_metric": null
+            },
+        "id": "EvVH6P7BgAA",
+		"lastUpdated": 0,
+		"lastUpdatedBy": null,
+		"metric": "cpu.utilization"
+       },
+       ....
+       ....
+	]
+}
+*/
+func getMetricsFromPayloads(metricData interface{}, metadata interface{}) (map[string]watcher.Metric, error) {
+	keyHostMap := make(map[string]string)
+	hostMetricMap := make(map[string]watcher.Metric)
+	results := metadata.(map[string]interface{})["results"]
+	if results == nil {
+		return hostMetricMap, errors.New("unexpected payload: missing results field")
+	}
+
+	for _, v := range results.([]interface{}) {
+		id := v.(map[string]interface{})["id"].(string)
```

Comment:
It's good practice to make sure the type assertion was successful using the tuple return (val,ok) as you do a few lines below.

I think this has the potential to crash the program since it's possible that v cannot be converted to a map type, and therefore couldn't have an item `["id"]`

Reply:
Yes you are right this is sloppy. Let me fix it.